File: collect_app/src/main/java/org/odk/collect/android/views/MultiClickSafeTextInputEditText.java
package org.odk.collect.android.views;

import android.content.Context;
import android.util.AttributeSet;

import androidx.annotation.NonNull;

import com.google.android.material.textfield.TextInputEditText;

import org.odk.collect.android.utilities.MultiClickGuard;

public class MultiClickSafeTextInputEditText extends TextInputEditText {

    public MultiClickSafeTextInputEditText(@NonNull Context context) {
        super(context);
    }

    public MultiClickSafeTextInputEditText(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    public boolean performClick() {
        return MultiClickGuard.allowClick(getClass().getName()) && super.performClick();
    }
}

File: collect_app/src/main/java/org/odk/collect/android/views/MultiClickSafeTextInputEditText.kt
package org.odk.collect.android.views

import android.content.Context
import android.util.AttributeSet
import org.odk.collect.android.utilities.MultiClickGuard.allowClick
import com.google.android.material.textfield.TextInputEditText

class MultiClickSafeTextInputEditText : TextInputEditText {
    constructor(context: Context) : super(context)

    constructor(context: Context?, attrs: AttributeSet?) : super(
        context!!, attrs
    )

    override fun performClick(): Boolean {
        return allowClick(javaClass.name) && super.performClick()
    }
}
